### PR TITLE
add @http directive documentation to directives.md

### PR DESCRIPTION
**Pull request by onlybakam**
add http directive

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cli/graphql-transformer/directives.md [hunk 1]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+```
+
+**Combining the different components**
+
+You can use a combination of params, query, body, headers, and environments in your `@http` directive definition.
+
+Given the definition
+
+```graphql
+type Query {
```
Comment (attilah) [inline docs/cli/graphql-transformer/directives.md:2473]: ```type Mutation```

File: docs/cli/graphql-transformer/directives.md [hunk 2]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+which would send the following request:
+
+```
+GET /posts?sort=DESC&from=last-week&limit=5
+Host: www.example.com
+```
+
+**Request Body**
+
+The `@http` directive also allows you to specify the body of a request, which is used for `POST`, `PUT`, and `PATCH` requests. To create a new post, you can define the following.
```
Comment (attilah) [inline docs/cli/graphql-transformer/directives.md:2432]: We should call out that ```query``` is also generated in the resulting schema, that enables the usage of the same fields in query params as in body. Another difference in behavior is that when something is defined as an url parameter that will NOT be added to query or body. Perhaps this should be called out in an earlier section or in the end in a summary section?

File: docs/cli/graphql-transformer/directives.md [hunk 3]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+```graphql
+mutation update {
+  updatePost(
+    body: {title: "new title", description: "updated description", views: 100}
+    params: {id: "EXISTING_ID"}
+    query: {withComments: true}) {
+    id
+    title
+    description
+    comments {
```
Comment (attilah) [inline docs/cli/graphql-transformer/directives.md:2493]: Comments type should be added as well. Perhaps we should state that the following examples are using Post and Comment type with the following schema: ...... that way customers would know what can they expect.

File: docs/cli/graphql-transformer/directives.md [hunk 4]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+  views: 100
+}
+```
+
+**Advanced cases**
+
+In some cases, you may want to send a request based on existing field data. Take a scenario where you have a post and want to fetch comments associated with the post in a single query. You can fetch the comments based on the post id with the following definition. `$ctx.source` is a map that contains the resolution of the parent field (`Post`) and gives access to `id`.
+
+
+```graphql
```
Comment (attilah) [inline docs/cli/graphql-transformer/directives.md:2518]: We should include the getPost definition as well to make the example easier to try out just by copy and paste.

File: docs/cli/graphql-transformer/directives.md [hunk 5]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+Assuming that getPost retrieves a post with the id `POST_ID`, the comments field is resolved by sending this request to the endpoint
+
+```
+GET /posts/POST_ID/comments
+Host: www.example.com
+```
+
+Note that there is no check to ensure that the reference variable (here the post ID) exists. When using this technique, it is recommended to make sure the referenced field is non-null.
+
+### Generates
```
Comment (attilah) [inline docs/cli/graphql-transformer/directives.md:2550]: We should note that the current implementation of the @http transformer does not support the calling of other AWS services where SIG4 signature is required.

File: docs/cli/graphql-transformer/directives.md [hunk 6]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+```
+
+**Path Parameters**
+
+You can create dynamic paths by specifying params in the directive URL by using the special `:<param>` notation.
+
+Given the definition
+
+```graphql
+  getPost: Post @http(
```
[the remark this answers is not in the document]
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2375]: fixing

File: docs/cli/graphql-transformer/directives.md [hunk 7]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+  value: String
+}
+```
+
+### Usage
+
+The `@http` directive allows you to quickly connect HTTP or HTTPS endpoint to an AppSync API by creating an AWS AppSync HTTP resolver. To connect an AWS Lambda resolver, add the `@http` directive to a field in your `schema.graphql`. For example:
+
+```graphql
+type Query {
```
[the remark this answers is not in the document]
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2350]: good point.

File: docs/cli/graphql-transformer/directives.md [hunk 8]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+  listPosts: Post @http(
+    url: "https://www.example.com/posts"
+    headers: [{key: "X-Header", value: "X-Header-Value"}]
+    )
+}
+```
+
+**Path Parameters**
+
+You can create dynamic paths by specifying params in the directive URL by using the special `:<param>` notation.
```
[the remark this answers is not in the document]
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2370]: switching to "parameters" throughout and using `params` to refer to the specific input object.

File: docs/cli/graphql-transformer/directives.md [hunk 9]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+    url: "https://www.example.com/posts/:id"
+  )
+}
+```
+
+you can fetch a specific post
+
+```graphql
+query post {
+  getPost(params: {id: "POST_ID"}) {
```
[the remark this answers is not in the document]
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2385]: makes sense

File: docs/cli/graphql-transformer/directives.md [hunk 10]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+Host: www.example.com
+```
+
+**Request Body**
+
+The `@http` directive also allows you to specify the body of a request, which is used for `POST`, `PUT`, and `PATCH` requests. To create a new post, you can define the following.
+
+
+```graphql
+type Query {
```
[the remark this answers is not in the document]
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2436]: changing

File: docs/cli/graphql-transformer/directives.md [hunk 11]
```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
...
+
+Given the definition
+
+```graphql
+type Query {
+  listPosts(sort: String!, from: String!, limit: Int!): Post @http(
+    url: "https://www.example.com/posts"
+    )
+```
+
```
[the remark this answers is not in the document]
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2410]: adding.

File: docs/cli/graphql-transformer/directives.md (hunk 2 above)
Comment (onlybakam) [reply docs/cli/graphql-transformer/directives.md:2432]: adding

**Comment by brene** [comment]
@onlybakam Thanks for the contrib! Merging this PR to docs.amplify.aws/cli/graphql-transformer/directives

**Pull request merged**